src/ClientTrait.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud;
+
+use Google\Cloud\Storage\Connection\ConnectionInterface;
+
+/**
+ * Provides shared functionality between Clients.
+ */
+trait ClientTrait
```

Comment:
Does it warrant to be a Trait? It's just one variable and setter for now. Will you add some code in the future? Will you change the code here very often?